Automattic/jetpack
# Blocks: update logic to build frontend upgrade nudges

#### Changes proposed in this Pull Request:

* Previous work on this:
    * #13070 for the original implementation.
    * Updates in #16611 and #18239.
    * We talked about deprecating them entirely in #18054.
    * Discussion in 175-gh-view-design

In this PR, I'm proposing that we get rid of the previous implementation, and instead add HTML markup in the PHP generating the nudge. That will require updating that markup manually in the future, but does simplify the build quite a bit.

**Note**

- [ ] Removing the build process for the js and deleting `projects/plugins/jetpack/extensions/shared/components/style.scss` also stops the css file to be generated. We would need to find a way around that.

#### Jetpack product discussion

* Internal reference: p1646307142602639-slack-C02JJ910CNL

#### Does this pull request change what data or activity we track or use?

* No

#### Testing instructions:

* Start with a free site, connected to WordPress.com
* Enable upgrade nudges with `add_filter( 'jetpack_block_editor_enable_upgrade_nudge', '__return_true' );`
* Fake your site to be WoA here: https://github.com/Automattic/jetpack/blob/6535ed73efce172bfedf942aff6bb1e94a96a4f9/projects/plugins/jetpack/_inc/lib/admin-pages/class-jetpack-redux-state-helper.php#L166
* Go to Posts > Add New
* Insert a Premimum Content block, and some content in there.
* Save draft.
* Preview your draft.
* See that the nudge is displayed properly.
